File: Sort/Python/InsertionSort.py
def insertion(arr): # worst-case O(n2) and average complexity of О(n) and space complexity 1
    for i in range(1,len(arr)):
        j = i-1
        anchor = arr[i]
        while j >=0 and anchor < arr[j]:
            arr[j+1] = arr[j]
            j = j-1
        arr[j+1] = anchor
    return arr  

arr = [3,7,5,9,0,2,4]
insertion(arr)

File: Sort/Javascript/InsertionSort.js
function sort(arr) {
    for (let i = 1; i < arr.length; i++) {
        let j = i-1
        const anchor = arr[i]
        for (; j >= 0; j--) {
            if (arr[j] < anchor) break
            arr[j+1] = arr[j]
        }
        arr[j+1] = anchor
    }
    return arr
}

console.log(sort([3, 2, 6, 9, 10, 1, 56, 7, 11]))

module.exports = { sort }
